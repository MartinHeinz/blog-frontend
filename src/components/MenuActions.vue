<template>
    <span id="actions">
      <ul>
        <li>
            <router-link :to="url_previous">
                <v-hover>
                <v-icon :href="url_previous" @mouseover="prev_active = true" @mouseout="prev_active = false"
                        slot-scope="{ hover }"
                        :class="`${hover? 'icon-active': 'icon-inactive'}`">fas fa-chevron-left</v-icon>
                </v-hover>
            </router-link>
        </li>
        <li>
            <router-link :to="url_next">
                <v-hover>
                <v-icon :href="url_next" @mouseover="next_active = true" @mouseout="next_active = false"
                        slot-scope="{ hover }"
                        :class="`${hover? 'icon-active': 'icon-inactive'}`">fas fa-chevron-right</v-icon>
                </v-hover>
            </router-link>
        </li>
        <li>
            <v-hover>
            <v-icon :large=false href="#" @mouseover="top_active = true" @mouseout="top_active = false"
                    slot-scope="{ hover }"
                    :class="`${hover? 'icon-active': 'icon-inactive'}`">fas fa-chevron-up</v-icon>
            </v-hover>
        </li>
        <li>
            <v-hover>
            <v-icon href="#" @mouseover="share_active = true" @mouseout="share_active = false"
                    v-on:click="toggle_social_sharing"
                    slot-scope="{ hover }"
                    :class="`${hover? 'icon-active': 'icon-inactive'}`">fas fa-share-alt</v-icon>
            </v-hover>
        </li>
      </ul>

      <span class="info" v-show="prev_active">Previous post</span>
      <span class="info" v-show="next_active">Next post</span>
      <span class="info" v-show="top_active">Back to top</span>
      <span class="info" v-show="share_active">Share post</span>
    </span>
</template>

<script>

export default {
    name: 'MenuActions',
    props: {
        url_previous: String,
        url_next: String,
    },
    methods: {
        toggle_social_sharing() {
            this.$root.$emit('toggle_social_sharing');
        },
    },
    data() {
        return {
            prev_active: false,
            next_active: false,
            top_active: false,
            share_active: false,
        };
    },
};
</script>

<style>
    #actions {
        float: right;
        margin-top: 2rem;
        margin-right: 2rem;
        width: 100%;
        text-align: right;
    }
    #actions ul {
        display: block;
    }
    #actions .info {
        font-style: italic;
    }

    .icon-inactive {
        color: #c9cacc;
    }
    .icon-active {
        color: #d480aa;
        cursor: pointer;
    }
</style>
