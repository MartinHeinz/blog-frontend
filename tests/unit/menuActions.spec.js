import { createLocalVue, mount, RouterLinkStub } from '@vue/test-utils';
import Vuetify from 'vuetify';
import MenuActions from '@/components/MenuActions.vue';


describe('MenuActions.vue', () => {
    const localVue = createLocalVue();
    localVue.component('router-link', RouterLinkStub);
    localVue.use(Vuetify);
    const wrapper = mount(MenuActions, {
        localVue,
        propsData: {
            url_previous: '/blog/1',
            url_next: '/blog/3',
        },
    });
    it('renders correcly', () => {
        expect(wrapper.element).toMatchSnapshot();
    });
    it('renders correcly when icon is hovered', () => {
        wrapper.setData({ next_active: true });
        expect(wrapper.element).toMatchSnapshot();
    });
});
