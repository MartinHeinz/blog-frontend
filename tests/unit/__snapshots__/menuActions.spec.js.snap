// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuActions.vue renders correcly 1`] = `
<span
  id="actions"
>
  <ul>
    <li>
      <a>
        <i
          aria-hidden="true"
          class="v-icon fas fa-chevron-left theme--light icon-inactive"
          href="/blog/1"
        />
      </a>
    </li>
     
    <li>
      <a>
        <i
          aria-hidden="true"
          class="v-icon fas fa-chevron-right theme--light icon-inactive"
          href="/blog/3"
        />
      </a>
    </li>
     
    <li>
      <i
        aria-hidden="true"
        class="v-icon fas fa-chevron-up theme--light icon-inactive"
        href="#"
      />
    </li>
     
    <li>
      <i
        aria-hidden="true"
        class="v-icon v-icon--link  fas fa-share-alt theme--light icon-inactive"
        href="#"
      />
    </li>
  </ul>
   
  <span
    class="info"
    style="display: none;"
  >
    Previous post
  </span>
   
  <span
    class="info"
    style="display: none;"
  >
    Next post
  </span>
   
  <span
    class="info"
    style="display: none;"
  >
    Back to top
  </span>
   
  <span
    class="info"
    style="display: none;"
  >
    Share post
  </span>
</span>
`;

exports[`MenuActions.vue renders correcly when icon is hovered 1`] = `
<span
  id="actions"
>
  <ul>
    <li>
      <a>
        <i
          aria-hidden="true"
          class="v-icon fas fa-chevron-left theme--light icon-inactive"
          href="/blog/1"
        />
      </a>
    </li>
     
    <li>
      <a>
        <i
          aria-hidden="true"
          class="v-icon fas fa-chevron-right theme--light icon-inactive"
          href="/blog/3"
        />
      </a>
    </li>
     
    <li>
      <i
        aria-hidden="true"
        class="v-icon fas fa-chevron-up theme--light icon-inactive"
        href="#"
      />
    </li>
     
    <li>
      <i
        aria-hidden="true"
        class="v-icon v-icon--link  fas fa-share-alt theme--light icon-inactive"
        href="#"
      />
    </li>
  </ul>
   
  <span
    class="info"
    style="display: none;"
  >
    Previous post
  </span>
   
  <span
    class="info"
    style=""
  >
    Next post
  </span>
   
  <span
    class="info"
    style="display: none;"
  >
    Back to top
  </span>
   
  <span
    class="info"
    style="display: none;"
  >
    Share post
  </span>
</span>
`;
